gax-grpc/src/main/java/com/google/api/gax/grpc/CallContext.java

```diff
@@ -67,11 +71,41 @@ public CallOptions getCallOptions() {
     return callOptions;
   }
 
-  public CallContext withChannel(Channel channel) {
-    return new CallContext(channel, this.callOptions);
+  public GrpcCallContext withChannel(Channel channel) {
+    return new GrpcCallContext(channel, this.callOptions);
   }
 
-  public CallContext withCallOptions(CallOptions callOptions) {
-    return new CallContext(this.channel, callOptions);
+  public GrpcCallContext withCallOptions(CallOptions callOptions) {
+    return new GrpcCallContext(this.channel, callOptions);
+  }
+
+  @Override
+  public <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> newUnaryCallable(
+      UnaryCallable<RequestT, ResponseT> unaryCallable) {
+    return new EntryPointUnaryCallable<>(unaryCallable, this);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    GrpcCallContext that = (GrpcCallContext) o;
+
+    if (channel != null ? !channel.equals(that.channel) : that.channel != null) {
+      return false;
+    }
+    return callOptions.equals(that.callOptions);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = channel != null ? channel.hashCode() : 0;
+    result = 31 * result + callOptions.hashCode();
+    return result;
```

Comment:
It was just for testing purposes. Good idea on `Objects.hash` & `Objects.equal`. The code here was generated by IntelliJ.